src/main/java/io/zold/api/Copies.java

```diff
@@ -24,37 +24,78 @@
 package io.zold.api;
 
 import io.zold.api.Copies.Copy;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.cactoos.collection.CollectionOf;
 import org.cactoos.iterable.IterableEnvelope;
 import org.cactoos.iterable.IterableOf;
+import org.cactoos.iterable.Joined;
 import org.cactoos.iterable.Mapped;
 
 /**
  * Multiple copies of a Wallet.
- *
  * @since 1.0
- * @todo #41:30min The constructor of Copies should be implemented
- *  to retrieve the Wallet with the provided id in the provided remotes and
- *  group them by equivalent wallet (i.e., equivalent content) along
- *  with their remotes as explained in the whitepaper. A unit test should be
- *  added to validate this behaviour. The unit test
- *  NetworkTest.pullIsNotYetImplemented() should also be removed and replaced
- *  with a real test.
  */
 public final class Copies extends IterableEnvelope<Copy> {
 
     /**
      * Ctor.
-     *
      * @param id Id of the wallet to pull.
      * @param remotes Remote nodes.
      */
     Copies(final long id, final Iterable<Remote> remotes) {
-        super(() -> new IterableOf<>(new Copy(new Wallet.Fake(id), remotes)));
+        super(() -> copies(id, remotes));
+    }
+
+    /**
+     * Builds copies from remotes.
+     * @param id Wallet's id
+     * @param remotes List of remotes
+     * @return Iterable Iterable of Copy
+     * @throws IOException If fails
+     */
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
+    private static Iterable<Copy> copies(final long id,
+        final Iterable<Remote> remotes) throws IOException {
+        final List<Copy> copies = new ArrayList<>(0);
```

Comment:
@Vatavuk let's not initialize with size zero, this will force unnecessary reallocation internally for the first element. Let's use as small sensible default, like 10.